blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -67,28 +67,30 @@ public static Socket connect(String host, int port, String username, String pass
     }
 
     public static void send(Message message, OutputStream outputStream) throws IOException {
-        log.trace("Writing on: " + outputStream);
-        StringBuffer toSend = new StringBuffer(message.getCommand().toString());
-        toSend.append("\n");
-
-        Iterator<String> keys = message.getHeaders().keySet().iterator();
-        while (keys.hasNext()) {
-            String key = keys.next();
-            String value = message.getHeaders().get(key);
-            toSend.append(key);
-            toSend.append(":");
-            toSend.append(value);
+        synchronized (outputStream) {
```

Comment:
I have not studied the issue in any detail but just looking at the code changes I don't think you need to synchronize on the whole method, maybe just on the bit that writes to the socket.

In fact, if you assemble the data and do a single write I would expect OutputStream to write it out in a single go - ie I don't think you need any synchronization at all.